R/RLearner_classif_fdaFDboost.R

```diff
@@ -0,0 +1,116 @@
+      # makeDiscreteLearnerParam(id = "baselearner", values = c("bbs", "bols")),  # we don't use "btree" in FDboost
+      makeIntegerLearnerParam(id = "knots", default = 10L, lower = 1L),  # determine the number of knots of splines, does not matter once there is sufficient number of knots, 30,40, 50 for example
+      makeIntegerLearnerParam(id = "degree", default = 3L, lower = 1L),  # degree of the b-spline
+      makeIntegerLearnerParam(id = "differences", default = 1L, lower = 1L),  # degree of the penalty
+      makeLogicalLearnerParam(id = "bsignal.check.ident", default = FALSE, tunable = FALSE)  # identifiability check by testing matrix degeneracy
+    ),
+    par.vals = list(family = "Binomial"),
+    properties = c("functionals", "numerics", "twoclass"),
+    name = "Functional linear array classification boosting",
+    short.name = "FDboost",
+    note = "Only allow one base learner for functional covariate and one base learner for scalar covariate, the parameters for these base learners are the same. Also we currently do not support interaction between scalar covariates"
+  )
+}
+
+#' @export
+trainLearner.classif.fdaFDboost = function(.learner, .task, .subset, .weights = NULL, mstop = 100L,
+  knots = 10L, df = 4L, bsignal.check.ident = FALSE, degree = 3L, differences = 1L, Binomial.link = "logit",
+  nu = 0.1, family = "Gaussian", custom.family.definition = NULL, nuirange = c(0, 100), d = NULL, ...) {
+
+  family = switch(family,
+    Binomial = mboost::Binomial(link = Binomial.link),
+    AdaExp = mboost::AdaExp(),
+    AUC = mboost::AUC(),
+    #PropOdds = mboost::PropOdds(nuirange = nuirange, offrange = offrange),
+    custom.family = custom.family.definition)
+
+  ctrl = learnerArgsToControl(mboost::boost_control, mstop, nu)
+
+  suppressMessages({d = getTaskData(.task, functionals.as = "dfcols")})
+  m = getTaskData(.task, functionals.as = "matrix")
+  tn = getTaskTargetNames(.task)
+
+  formula.terms = namedList()
+  mat.list = namedList(getTaskFeatureNames(.task))
+
+  # Treat functional covariates
+  if (hasFunctionalFeatures(m)) {
+    fdns = colnames(getFunctionalFeatures(m))
+    # later on, the grid elements in mat.list should have suffix ".grid"
+    fdg = namedList(fdns)
+    fd.grids = lapply(fdns, function(name) seq_len(ncol(m[, name])))
+    names(fd.grids) = fdns
+    fdg = setNames(fd.grids, stri_paste(fdns, ".grid"))
+    # setup mat.list: for each func covar we add its data matrix and its grid. and once the target col
+    # also setup charvec of formula terms for func covars
+    mat.list = namedList(fdns)
+    #formula.terms = setNames(character(length = fdns))
+    formula.terms = namedList(fdns)
+    # for each functional covariate
+    for (fdn in fdns) {
```

Comment:
Can we write a function which we use for both (regr/classif) learners? Do you think that would make sense?